src/elf/dyn.rs

```diff
@@ -497,12 +502,9 @@ macro_rules! elf_dyn_std_impl {
                     if phdr.p_type == PT_DYNAMIC {
                         let filesz = phdr.p_filesz as usize;
                         let dync = filesz / SIZEOF_DYN;
-                        let mut bytes = vec![0u8; filesz];
+                        let mut dyns = vec![Dyn::default(); dync];
                         try!(fd.seek(Start(phdr.p_offset as u64)));
-                        try!(fd.read(&mut bytes));
-                        let bytes = unsafe { slice::from_raw_parts(bytes.as_ptr() as *mut Dyn, dync) };
-                        let mut dyns = Vec::with_capacity(dync);
-                        dyns.extend_from_slice(bytes);
+                        try!(fd.read(dyns.as_mut_bytes()));
                         dyns.dedup();
```

Comment:
Is this dedup() actually useful? Seems like a waste of time to me, and besides, it makes the parsed state possibly different from what's in the file.